src/arm/cache.c

```diff
@@ -1341,7 +1341,8 @@ void cpuinfo_arm_decode_cache(
 			 * information, please refer to the technical manuals
 			 * linked above
 			 */
-			const uint32_t min_l2_size_KB = uarch == cpuinfo_uarch_neoverse_v2 ? 1024 : 256;
+			const uint32_t min_l2_size_KB =
+				(uarch == cpuinfo_uarch_neoverse_v2 || midr_is_ampere_altra(midr)) ? 1024 : 256;
```

Comment:
I guess this is OK, and assume you have verified the sizes somehow. I also assume `CCSIDR_EL1, Current Cache Size ID Register` not readable from EL0.

Reply:
The Ampere Altra is a `cpu_uarch_neoverse_n1`, i.e. not a new `uarch`.

This function was currently setting the L2 cache size to the known minimum requirement for each `uarch`, regardless of the actual implementation.